go/ai/gen.go

```diff
@@ -106,30 +155,48 @@ type Message struct {
 
 type ModelInfo struct {
 	Label    string             `json:"label,omitempty"`
+	Stage    ModelInfoStage     `json:"stage,omitempty"`
 	Supports *ModelInfoSupports `json:"supports,omitempty"`
 	Versions []string           `json:"versions,omitempty"`
 }
 
+type ModelInfoStage string
+
+const (
+	ModelInfoStageFeatured   ModelInfoStage = "featured"
+	ModelInfoStageStable     ModelInfoStage = "stable"
+	ModelInfoStageUnstable   ModelInfoStage = "unstable"
+	ModelInfoStageLegacy     ModelInfoStage = "legacy"
+	ModelInfoStageDeprecated ModelInfoStage = "deprecated"
+)
+
 type ModelInfoSupports struct {
-	Constrained bool         `json:"constrained,omitempty"`
-	ContentType []string     `json:"contentType,omitempty"`
-	Context     bool         `json:"context,omitempty"`
-	Media       bool         `json:"media,omitempty"`
-	Multiturn   bool         `json:"multiturn,omitempty"`
-	Output      OutputFormat `json:"output,omitempty"`
-	SystemRole  bool         `json:"systemRole,omitempty"`
-	ToolChoice  bool         `json:"toolChoice,omitempty"`
-	Tools       bool         `json:"tools,omitempty"`
+	ContentType []string `json:"contentType,omitempty"`
+	Context     bool     `json:"context,omitempty"`
+	Media       bool     `json:"media,omitempty"`
+	Multiturn   bool     `json:"multiturn,omitempty"`
+	Output      []string `json:"output,omitempty"`
+	SystemRole  bool     `json:"systemRole,omitempty"`
+	ToolChoice  bool     `json:"toolChoice,omitempty"`
+	Tools       bool     `json:"tools,omitempty"`
 }
 
+type ModelInfoSupportsConstrained string
+
+const (
+	ModelInfoSupportsConstrainedNone    ModelInfoSupportsConstrained = "none"
+	ModelInfoSupportsConstrainedAll     ModelInfoSupportsConstrained = "all"
+	ModelInfoSupportsConstrainedNoTools ModelInfoSupportsConstrained = "no-tools"
+)
+
 // A ModelRequest is a request to generate completions from a model.
 type ModelRequest struct {
 	Config   any         `json:"config,omitempty"`
-	Context  []*Document `json:"context,omitempty"`
+	Docs     []*Document `json:"docs,omitempty"`
 	Messages []*Message  `json:"messages,omitempty"`
 	// Output describes the desired response format.
-	Output     *ModelRequestOutput `json:"output,omitempty"`
-	ToolChoice ToolChoice          `json:"toolChoice,omitempty"`
+	Output     *GenerateActionOutputConfig `json:"output,omitempty"`
```

Comment:
Even though it'll probably end up being identical, `ModelRequest.Output` is a different type than `GenerateActionOptions.Output`.